cli/lakecli/lakecli.go

```diff
@@ -0,0 +1,126 @@
+package lakecli
+
+import (
+	"context"
+	"encoding/hex"
+	"flag"
+	"fmt"
+	"strconv"
+
+	"github.com/brimdata/zed/api"
+	"github.com/brimdata/zed/cmd/zed/query"
+	"github.com/brimdata/zed/expr/extent"
+	"github.com/brimdata/zed/lake"
+	"github.com/brimdata/zed/lake/index"
+	"github.com/brimdata/zed/lake/journal"
+	"github.com/brimdata/zed/order"
+	"github.com/brimdata/zed/zbuf"
+	"github.com/brimdata/zed/zio"
+	"github.com/segmentio/ksuid"
+)
+
+type Flags interface {
+	SetFlags(fs *flag.FlagSet)
+	Quiet() bool
+	PoolName() string
+	Create(ctx context.Context) (Root, error)
+	Open(ctx context.Context) (Root, error)
+	OpenPool(ctx context.Context) (Pool, error)
+	CreatePool(ctx context.Context, layout order.Layout, thresh int64) (Pool, error)
+}
+
+type Root interface {
+	AddIndex(context.Context, []index.Index) error
+	CreatePool(context.Context, string, order.Layout, int64) (Pool, error)
+	OpenPool(context.Context, ksuid.KSUID) (Pool, error)
+	ScanPools(context.Context, zio.Writer) error
+	LookupPoolByName(context.Context, string) (*lake.PoolConfig, error)
+	RemovePool(context.Context, ksuid.KSUID) error
+	DeleteIndices(context.Context, []ksuid.KSUID) ([]index.Index, error)
+	LookupIndices(context.Context, []ksuid.KSUID) ([]index.Index, error)
+	ScanIndex(context.Context, zio.Writer, []ksuid.KSUID) error
+	Query(context.Context, zio.Writer, query.Includes, query.Includes) (zbuf.ScannerStats, error)
+}
+
+type Pool interface {
+	Config() lake.PoolConfig
+	// Add adds the contents of the reader to a pool. If the commit is not nil
+	// the data will be committed to the pool.
+	Add(ctx context.Context, r zio.Reader, commit *api.CommitRequest) (ksuid.KSUID, error)
+	Delete(ctx context.Context, tags []ksuid.KSUID, commit *api.CommitRequest) (ksuid.KSUID, error)
+	Commit(ctx context.Context, id ksuid.KSUID, commit api.CommitRequest) error
+	ScanLog(ctx context.Context, w zio.Writer, head, tail journal.ID) error
+	ScanStaging(ctx context.Context, w zio.Writer, ids []ksuid.KSUID) error
+	ScanSegments(ctx context.Context, w zio.Writer, at string, partitions bool, span extent.Span) error
+	Squash(ctx context.Context, ids []ksuid.KSUID, commit api.CommitRequest) (ksuid.KSUID, error)
+	Index(ctx context.Context, rules []index.Index, ids []ksuid.KSUID) (ksuid.KSUID, error)
+}
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#interfaces says interfaces generally belong in the package using values of that type. Can we follow that with these?

Reply:
I take it you are suggesting moving the above interfaces to ./cmd/zed/lake?

I tried doing this and it doesn't work because of return functions like: 
```
CreatePool(context.Context, string, order.Layout, int64) (Pool, error)
```
If I implement it like thus in lakecli:
```
CreatePool(context.Context, string, order.Layout, int64) (*LocalPool, error)
```
Go complains that it doesn't implement the Root interface. Open to suggestions...